index.js

```diff
@@ -163,6 +163,37 @@ module.exports = class Localdrive {
     }
   }
 
+  async * readdir (folder) {
+    const { keyname, filename: fulldir } = keyResolve(this.root, folder || '/')
+    const iterator = await readdir(fulldir)
+
+    if (!iterator) return
+
+    for await (const dirent of iterator) {
+      const key = unixPathResolve(keyname, dirent.name)
+
+      let suffix = key.slice(keyname.length)
+      const i = suffix.indexOf('/')
+      suffix = i === -1 ? suffix : suffix.slice(i + 1)
+
+      if (dirent.isDirectory()) {
+        let empty = true
+        for await (const entry of this.list(key)) { // eslint-disable-line
+          empty = false
+          break
+        }
+        if (empty) continue
```

Comment:
clearer to make a function, isEmptyDirectory for the above helper. Then you can return also